ssz/sedes/list.py

```diff
@@ -0,0 +1,85 @@
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes import (
+    address,
+    boolean,
+    hash32,
+    uint32,
+)
+
+
+class List:
+    """
+    A sedes for lists.
+    """
+    LENGTH_BYTES = 4
+
+    def __init__(self, sedes):
+        # This sedes object corresponds to each item of the list
+        self.sedes = sedes
+
+    def serialize(self, val):
+        if not isinstance(val, list):
+            raise SerializationError('Can only serialize lists', val)
+        # Make sure all items in list are of same type
+        if not all((type(item)) == (type(val[0])) for item in val):
```

Comment:
This must not be checked here, but in the loop where serialization happens. Also, indexing doesn't work for iterables in general, and even a list might be empty.